torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -232,6 +232,72 @@ Tensor norm_backward(Tensor grad, const Tensor& self, const optional<Scalar> & p
   return self_scaled * scale_v;
 }
 
+Tensor norm_jvp(
+  const Tensor& self_p, const Tensor& self_t,
+  const optional<Scalar> & p_,
+  Tensor norm,
+  IntArrayRef dim,
+  bool keepdim
+) {
+  // NB: currently norm_jvp is also reused for dist's jvp (which haas two differentiable inputs)
+  //     but self_t still cannot be a ZT because that would require both self_t and other_t to be ZT
+  TORCH_INTERNAL_ASSERT(!self_t._is_zerotensor());
+  size_t ndim = self_p.sizes().size();  // composite compliance?
+  double p = p_.value_or(2.0).toDouble();
+
+  if (p == 0.0) {
+    return at::zeros_like(norm);
+  } else if (p == 1.0) {
+    auto result = self_p.sgn();
+    result = areAnyTensorSubclassLike({self_t}) ? result.mul(self_t.conj()) : result.mul_(self_t.conj());
+    result = at::real(result);
+    return result.sum(dim, keepdim);
+  } else if (p == 2.0) {
+    auto result = self_p.sgn();
+    result = areAnyTensorSubclassLike({self_t}) ? result.mul(self_t.conj()) : result.mul_(self_t.conj());
+    result = at::real(result);
+    result *= self_p.abs();
+    result = result.sum(dim, keepdim);
+    return result.div_(norm);
```

Comment:
You're missing the `masked_fill_` here no? If norm has any `0` value this will go to Nan.
Is this case actually tested in OpInfo or not? IIRC we have a one off test for this.

Reply:
Yes, good catch, and also yes it should've been tested. The one-off test doesn't actually test forward AD though. I've updated the OpInfo now to test this case.

By the way, that one-off test was actually removed with the intention of subsuming it into the OpInfo, so... there was some time where it wasn't being tested for backward either. Everything should be tested now though.

Reply:
I can also see an argument for keeping one-off tests for subgradient tests: being able to see what the expected values actually are instead of having to reason about what the gradcheck is doing. The flipside is that if we have both forward AD and backward AD tests, then it starts getting more annoying to check both. So, I'm thinking as long as the test computes the subgradient as expected by the gradcheck, we can just let OpInfo handle it.

Reply:
Yes having OpInfo handling it is the way to go for this one.